src/telemetry/index.ts

```diff
@@ -0,0 +1,57 @@
+import TelemetryReporter from "@vscode/extension-telemetry";
+import * as vscode from "vscode";
+import { provideSingleton } from "../utils";
+
+@provideSingleton(TelemetryService)
+export class TelemetryService implements vscode.Disposable {
+  private telemetryReporter: TelemetryReporter = new TelemetryReporter(
+    "50598369-dd83-4f9a-9a65-ca1fa6f1785c",
+  );
+
+  sendTelemetryEvent(
+    eventName: string,
+    properties?: { [key: string]: string },
+    measurements?: { [key: string]: number },
+  ) {
+    if (this.isTelemetryEnabled()) {
+      this.telemetryReporter.sendTelemetryEvent(
+        eventName,
+        properties,
+        measurements,
+      );
+    }
+  }
+
+  sendTelemetryError(
+    eventName: string,
+    error?: unknown,
+    properties?: { [key: string]: string },
+    measurements?: { [key: string]: number },
+  ) {
+    if (this.isTelemetryEnabled()) {
+      this.telemetryReporter.sendTelemetryErrorEvent(
+        eventName,
+        {
+          ...properties,
+          stack:
+            error !== undefined && error instanceof Error
+              ? error.stack
+              : undefined,
+        },
+        measurements,
+      );
+    }
+  }
+
+  private isTelemetryEnabled(): boolean {
+    return (
+      vscode.workspace
+        .getConfiguration()
+        .get<boolean>("telemetry.enableTelemetry") !== false
+    );
+  }
```

Comment:
This should not be overridden. isTelemetryEnabled flag should already be handled as part of the telemetry package.
I think originally the idea was for us to have an additional flag so that people can turn off telemetry just for this plugin and not need to turn it off for everything. The config name needs to be under dbt. asked @anandgupta42  to update his branch with an example.

Reply:
reference from vscode docs: https://code.visualstudio.com/api/extension-guides/telemetry